### ISSUE TITLE
`ts/no-unnecessary-type-assertion` rule not applying when configured in `typescript.overridesTypeAware`

### ISSUE THREAD

**Issue by blouflashdb** [antfu/eslint-config#777]
### Describe the bug

I encountered a problem when trying to override/enable the type-aware rule `ts/no-unnecessary-type-assertion` using the `typescript.overridesTypeAware` option in the `antfu()` factory.

The rule is a type-aware rule, and I have configured it alongside other type-aware rules, but it does not appear to be active according to the ESLint Config Inspector or in my IDE (VS Code).

#### Expected Behavior

The rule `ts/no-unnecessary-type-assertion` should be enabled with `error` severity, applying to all type-aware files configured by `antfu/eslint-config`.

#### Current Behavior

The rule is not enabled and does not report any errors in my project, even for code that should clearly trigger it (e.g., a redundant type assertion).

It is not visible in the configuration for TypeScript files when inspecting the ESLint config with tools like the ESLint Config Inspector.

#### Configuration

This is my `eslint.config.mjs`:

```javascript
// @ts-check
import antfu from '@antfu/eslint-config'

export default antfu({
  typescript: {
    overridesTypeAware: {
      'ts/no-unnecessary-type-assertion': 'error',
    },
  },
})
```

### Reproduction

https://github.com/blouflashdb/minimal-repro

### System Info

*   **`antfu/eslint-config` version:** 6.2.0
*   **ESLint version:** 9.38.0
*   **Node version:** 22.20.0
*   **Operating System:** Windows 11

### Used Package Manager

npm

### Validations

- [x] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [x] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

**Comment by blouflashdb** [comment]
`tsconfigPath: "./tsconfig.json",` was missing in the config. I thought that this is the default value when typescript is enabled.

**Comment by christopher-buss** [comment]
@blouflashdb Type-aware lints are not enabled by default (even when you enable TypeScript) unless you supply a tsconfig path.
